lib/customer.rb

```diff
@@ -0,0 +1,40 @@
+require 'csv'
+
+
+class Customer 
+    
+    attr_reader :id 
+    attr_accessor :email, :address
+    
+    def initialize(id,email,address)
+        @id = id
+        @email = email
+        @address = address
+    end 
+    
+    def self.all
+        data = CSV.read('data/customers.csv')
+        customers = data.map do |person|
+            address = {street: person[2], city: person[3], state: person[4], zip: person[5]}
+            self.new(person[0].to_i, person[1], address)
+        end 
+        customers     
+    end 
+    
+    
+    def self.find (id)
```

Comment:
You shouldn't have spaces between a method name and its argument list.

```suggestion
    def self.find(id)
```

That's where this warning is coming from:
```
/Users/kaida/Developer/cohort-12/grocery-store/lib/customer.rb:25: warning: parentheses after method name is interpreted as an argument list, not a decomposed argument
```